Library/Homebrew/utils/pypi.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module PyPI
+  module_function
+
+  # Ensure pipgrip is installed
+  def ensure_pipgrip_installed
+    odie '"pipgrip" must be installed (`brew install pipgrip`)' unless Formula["pipgrip"].any_version_installed?
+  end
+
+  # Get name, url, and version for a given pypi package
+  def get_pypi_info(package, version)
+    metadata_url = "https://pypi.org/pypi/#{package}/#{version}/json"
+    out, _, status = curl_output metadata_url, "--location"
+
+    return unless status.success?
+
+    begin
+      json = JSON.parse out
+    rescue JSON::ParserError
+      return
+    end
+
+    sdist = json["urls"].find { |url| url["packagetype"] == "sdist" }
+    [json["info"]["name"], sdist["url"], sdist["digests"]["sha256"]]
+  end
+
+  def update_python_resources!(formula, version = nil, print_only: false, silent: false,
+                               ignore_non_pypi_packages: false)
+    # PyPI package name isn't always the same as the formula name. Try to infer from the URL.
+    pypi_name = if formula.stable.url.start_with?("https://files.pythonhosted.org/packages/")
+      File.basename(formula.stable.url).match(/^(.+)-[a-z\d.]+$/)[1]
+    else
+      formula.name
+    end
+
+    version ||= formula.version
+
+    if get_pypi_info(pypi_name, version).blank?
+      odie "\"#{pypi_name}\" at version #{version} is not available on PyPI." unless ignore_non_pypi_packages
+      return
+    end
+
+    non_pypi_resources = formula.resources.reject do |resource|
+      resource.url.start_with? "https://files.pythonhosted.org/packages/"
```

Comment:
This string is used a few times. Would be good to make it a module constant instead.